fabric-network/src/impl/query/singlequeryhandler.js

```diff
@@ -26,38 +25,29 @@ class SingleQueryHandler {
 
 		const startPeerIndex = this._currentPeerIndex;
 		const errorMessages = [];
-		const options = {requestTimeout: 3000};
-		// use the timeout as the requestTimeout or let default
-		if (Number.isInteger(this._options.timeout)) {
-			options.requestTimeout = this._options.timeout * 1000; // in ms;
-		}
 
 		for (let i = 0; i < this._peers.length; i++) {
 			const peerIndex = (startPeerIndex + i) % this._peers.length;
 			this._currentPeerIndex = peerIndex;
+
 			const peer = this._peers[peerIndex];
-			logger.debug('%s - query sending to peer %s', method, peer.name);
-			try {
-				const results = await query.send({targets:[peer]}, options);
-				if (results && results.errors && results.errors.length > 0) {
-					logger.error('%s - problem with query to peer %s error:%s', method, peer.name, results.errors[0]);
-					// since only one peer, only one error
-					errorMessages.push(results.errors[0].message);
-				} else if (results && results.responses && results.responses[0].response.status < 400) {
-					logger.debug('%s - peer response status %s', method, results.responses[0].response.status);
-					return results.responses[0].response.payload;
-				} else if (results && results.responses && results.responses[0].response.message) {
-					logger.debug('%s - peer response status: %s message: %s', method,
-						results.responses[0].response.status,
-						results.responses[0].response.message);
-					errorMessages.push(results.responses[0].response.message);
+			logger.debug('%s - sending to peer %s', method, peer.name);
+
+			const results = await query.send([peer]);
+			if (results && results[peer.name]) {
```

Comment:
We (and client developers) really shouldn't need to do this checking. The `query.evaluate()` should guarantee to get a result containing results for all the specified peers.

Reply:
fixed